carapace-server/src/main/java/org/carapaceproxy/server/certificates/DynamicCertificatesManager.java

```diff
@@ -273,11 +284,10 @@ private void certificatesLifecycle() {
                 .sorted((e1, e2) -> e1.getKey().compareTo(e2.getKey()))
                 .map(e -> e.getValue())
                 .collect(Collectors.toList());
-        for (CertificateData data : _certificates) {
+        for (CertificateData cert : _certificates) {
             boolean updateCertificate = true;
-            final String domain = data.getDomain();
+            final String domain = cert.getDomain();
             try {
-                CertificateData cert = loadOrCreateDynamicCertificateForDomain(domain, data.isWildcard(), false, data.getDaysBeforeRenewal());
```

Comment:
We skip retriving the certificate from db which is already up-to-date. (all certs are reloaded as soon as a node triggers fire-cache request)